### PR TITLE
[FEATURE] V2-to-V3 Upgrade/Migration

### PR DESCRIPTION
**Pull request by alexsherstinsky**
**Broadly speaking, this PR can be thought of in 3 parts:**

1. How-to-guide on Migration from V2 to V3, which can be found here : https://deploy-preview-3592--niobium-lead-7998.netlify.app/docs/guides/miscellaneous/migration_guide
2.  CLI changes and tests associated with migration script (specifically `project check-config` and `project upgrade`)
3. Example configuration that can be used to follow the steps in the migration guide and run the migration. 
    - found here: `great_expectations/tests/test_fixtures/configuration_for_testing_v2_v3_migration/`
- JIRA: GREAT-200
- JIRA: GREAT-302/GREAT-303

**The review process will involve multiple steps, including**
- [x] Friction logging for How-to-guide (do the steps work? did you get stuck?)
- [x] Review of the content of the How-to-guide (does there need to be changes in phrasing, or additional information / information taken out?)
- [x] Code review of the CLI changes and tests
Thank you in advance for this 🙇🏼 

**So, what is in the Example configuration?**
- The `great_expectations/tests/test_fixtures/configuration_for_testing_v2_v3_migration/` contains a complete-and-working V2 Configuration and V3 Configuration that can be used to run a migration. 

- Test File : `Titanic.csv` that is found in the base directory and is referred to by both configurations
- V2 configuration in `v2/great_expectations/` folder 
   - Checkpoint named `test_v2_checkpoint`
      - uses `LegacyCheckpoint` class
      - uses `batch_kwargs` 
      - uses Validation Operator `action_list_operator`
      - references `Titanic.csv` testfile
   - `great_expectations.yml`
       - uses `config_version: 2.0` 
       - uses v2-datasource
       - uses `validation_operators`
       - no checkpoint stores

-  V3 configuration and `v3/great_expectations/` folder 
   - Checkpoint named `test_v3_checkpoint`
      - uses `Checkpoint` class
      - uses `batch_request` 
      - references `Titanic.csv` testfile
   - `great_expectations.yml`
       - uses `config_version: 3.0` 
       - uses v3-datasource
       - uses `CheckpointStore`

- Both checkpoints can be run on their own (using `running_checkpoint.sh` which is included for convenient) without any changes, and can be used as references for V2 and V3 configurations. 

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: docs/guides/miscellaneous/migration_guide.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ Would you like to proceed with the project upgrade? [Y/n]:
 If you select `Y`, then the following upgrade will automatically occur:
 
 ```bash
-Upgrading project...
+pgrading project...
```
Comment (anthonyburdi) [inline docs/guides/miscellaneous/migration_guide.md]:
```suggestion
Upgrading project...
```

File: docs/guides/miscellaneous/migration_guide.md [hunk 2]
```diff
@@ -182,7 +186,19 @@ Migrating Datasource configurations that contain connections to the cloud or dat
...
+In Great Expectations version 0.13.7, we introduced an improved Checkpoints feature, which allowed Checkpoints to utilize features the V3 API. As a result, Checkpoints are now able to [filter and sort batches from configured datasources](/docs/guides/connecting_to_your_data/how_to_get_a_batch_of_data_from_a_configured_datasource), [introspect and partition tables as batches](/docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql), with multi-batch Expectations soon to come.  As part of these design improvements, Validation Operators (originally located in the `great_expectations.yml` file) were combined into the Checkpoint configuration itself as `action_list` items.
```
Comment (anthonyburdi) [inline docs/guides/miscellaneous/migration_guide.md:189]: ❤️

File: docs/guides/miscellaneous/migration_guide.md (hunk 1 above)
Comment (talagluck) [inline docs/guides/miscellaneous/migration_guide.md]: This looks like a typo

File: docs/guides/miscellaneous/migration_guide.md [hunk 3]
```diff
@@ -163,11 +163,15 @@ The V2-style Datasource has:
...
   - Data-specific DataConnectors, like the `InferredAssetFilesystemDataConnector` in our example below.
 
-For reference, a [Data Connector](../../reference/datasources) facilitates access to an external data store, such as a database, filesystem, or cloud storage and the [Execution Engine](../../reference/execution_engine) is responsible for providing the computing resources that will be used to actually perform validation. For more information on these V3-style components with additional examples, please refer to the [Datasources Documentation](../../reference/datasources).
+:::note Note Datasource in V3
+
+One exception to the datatype-agnostic Datasource in the V3 API is the SimpleSqlalchemyDatasource, which combines functionality of the Datasource and ExecutionEngine to enable [database introspection and partitioning](/docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql). More examples on using the SimpleSqlalchemyDatasource can be found [here](/docs/guides/connecting_to_your_data/how_to_configure_a_dataconnector_to_introspect_and_partition_tables_in_sql).
```
Comment (talagluck) [inline docs/guides/miscellaneous/migration_guide.md:172]: ♥️

File: docs/guides/miscellaneous/migration_guide.md (hunk 1 above)
Comment (Shinnnyshinshin) [reply docs/guides/miscellaneous/migration_guide.md]: my fault 😅  just pushed a fix

**Review by talagluck — APPROVED** [review]
I just reviewed the `migration_guide`, but it looks awesome - thanks for the updates! Just found one minor typo.

File: docs/guides/miscellaneous/migration_guide.md (hunk 3 above)
Comment (Shinnnyshinshin) [reply docs/guides/miscellaneous/migration_guide.md:172]: ❤️

**Review by anthonyburdi — APPROVED** [review]
Minor tweak but otherwise the latest changes LGTM!

File: docs/guides/miscellaneous/migration_guide.md (hunk 1 above)
Comment (anthonyburdi) [reply docs/guides/miscellaneous/migration_guide.md]: Already fixed - nice 😄

Comment (Shinnnyshinshin) [reply docs/guides/miscellaneous/migration_guide.md]: I think you and @talagluck both caught the same typo 😅  this has been resolved with the most recent push

**Pull request merged**